sdk/python/feast/infra/provider.py

```diff
@@ -202,6 +212,13 @@ def _get_requested_feature_views_to_features_dict(
 
         if not found:
             raise ValueError(f"Could not find feature view from reference {ref}")
+
+    if feature_names_only and len(feature_collision_set) > 0:
+        err = ", ".join(x for x in feature_collision_set)
+        raise ValueError(
```

Comment:
Same comment as above (https://github.com/feast-dev/feast/pull/1641/files#r650120378) for using a custom error type like `FeatureNameCollisionError` (Those can be set in [sdk/python/feast/errors.py](https://github.com/feast-dev/feast/blob/master/sdk/python/feast/errors.py))